tests/test_georaster.py

```diff
@@ -209,46 +209,83 @@ def test_inters_img(self):
 
     def test_interp(self):
 
+        # FIRST, we try on a Raster with a Point interpretation in its "AREA_OR_POINT" metadata: values interpolated
+        # at the center of pixel
         r = gr.Raster(datasets.get_path("landsat_B4"))
+        assert r.ds.tags()['AREA_OR_POINT'] == 'Point'
 
         xmin, ymin, xmax, ymax = r.ds.bounds
 
-        # testing interp, find_value, and read when it falls right on the coordinates
+        # We generate random points within the boundaries of the image
         xrand = (np.random.randint(low=0, high=r.ds.width, size=(10,))
                  * list(r.ds.transform)[0] + xmin + list(r.ds.transform)[0]/2)
         yrand = (ymax + np.random.randint(low=0, high=r.ds.height, size=(10,))
                  * list(r.ds.transform)[4] - list(r.ds.transform)[4]/2)
         pts = list(zip(xrand, yrand))
+        # Get decimal indexes, those should all be .5 because values refer to the center
+        i, j = r.xy2ij(xrand, yrand,op=np.float32,shift_area_or_point=True)
+        assert np.all(i % 1 == 0.5)
+        assert np.all(j % 1 == 0.5)
+
+        # now we calculate the mean of values in each 2x2 slices of the data, and compare with interpolation at order 1
+        list_z_ind = []
+        img = r.data
+        for k in range(len(xrand)):
+            # 2x2 slices
+            z_ind = np.mean(img[0, slice(int(np.floor(i[k])),int(np.ceil(i[k]))+1), slice(int(np.floor(j[k])),int(np.ceil(j[k]))+1)])
+            list_z_ind.append(z_ind)
+
+        # order 1 interpolation
+        rpts = r.interp_points(pts,order=1)
+        # the values interpolated should be equal
+        assert np.array_equal(np.array(list_z_ind,dtype=np.float32),rpts,equal_nan=True)
+
+        # Test there is no failure with random coordinates (edge effects, etc)
+        xrand = np.random.uniform(low=xmin, high=xmax, size=(1000,))
+        yrand = np.random.uniform(low=ymin, high=ymax, size=(1000,))
+        pts = list(zip(xrand, yrand))
+        rpts = r.interp_points(pts)
+
+        # SECOND, test after a crop: the Raster now has an Area interpretation, those should fall right on the integer
```

Comment:
It's weird that crop changes the area_or_point attribute...

Reply:
You think it comes from `_update()` ?

Reply:
You can test...

Reply:
If you can't find out why, maybe add as an issue.

Reply:
Added #127